### ISSUE TITLE
Game crashes at random times

### ISSUE THREAD

**Issue by krishraghuram** [ValveSoftware/Dota-2#2664]
#### Your system information

* System information from steam (`Steam` -> `Help` -> `System Information`) in a [gist](https://gist.github.com/): https://gist.github.com/krishraghuram/d5781c4530ba0d32535bff22b4221d99

Steam Runtime Diagnostics: https://gist.github.com/krishraghuram/af1ab0b7030443bac4802879f1295bd1
* Have you checked for system updates?: No
* Are you using the latest stable video driver available for your system? Yes
* Have you verified the game files?: Yes

#### Please describe your issue in as much detail as possible:
Describe what you _expected_ should happen and what _did_ happen. Please link any large pastes as a [Github Gist](https://gist.github.com/).

Game crashes at random times during gameplay. Would really love to provide more information, but I cant really narrow down as to what's causing the crash. Would love to provide log files, but I'm not sure where I should look for logs.

#### Steps for reproducing this issue:

1. Play the game
2. Game crashes at random times

**Comment by formidabilus** [comment]
Same here, on Fedora 39 with Wayland and Nvidia.

**Comment by headlesscyborg** [comment]
Not sure if I'm experiencing the same issue but for me, it's guaranteed that the game crashes on alt+tab, changing volume or doing anything that somehow interacts with the desktop.
Creating a guild (loading guild logo from the disk and pressing "Create guild") also leads to a crash. It's stable as long as I just play without interrupting the game window.

Arch Linux, RX 7900 XT, GNOME Wayland

EDIT:
- guild logo crash might be related to this https://github.com/ValveSoftware/Dota-2/issues/2520
- alt tab workaround might be playing in borderless window

**Comment by formidabilus** [comment]
> Replying to https://github.com/ValveSoftware/Dota-2/issues/2664#issuecomment-1951161846

Yes, when changing focus from Dota2 to anything else it crashes at random times. For me it crashed even when I tried to change to borderless window and if I remember correctly it crashed after that, too.  

Using Fedora 39, Gnome Wayland and Nvidia 3070Ti.

**Comment by estavrionto** [comment]
disabling the "bring dota 2 to foreground" options in advanced options seems to have helped me with this.
